product_docs/docs/migration_toolkit/55/07_invoking_mtk/mtk_command_options_in_file/creating_txt_file.mdx

```diff
@@ -0,0 +1,108 @@
+
+### Use a line per option
+
+Use a new line for each option. Use a line per key-value pair.
+
+| Option in CLI                                                  | Supported syntax for the option in file                       |
+|----------------------------------------------------------------|---------------------------------------------------------------|
+| `-tables hr -allViews` <br/> `-excludeViews all_emp,acct_list` | tables hr <br/> allViews <br/> excludeViews all_emp,acct_list |
+
+### Specify values for an option in a single or multiple lines
+
+You can add a single line with an option and multiple values for that option.
+
+You can add several lines for the same option, each time with a different value.
+
+And you can combine both approaches, because MTK will combine all lines that have the same option.
+
+| Option in CLI            | Supported syntax for the option in file         |
+|--------------------------|-------------------------------------------------|
+| `-tables TAB1,TAB2,TAB3` | tables TAB1 <br/> tables TAB2 <br/> tables TAB3 |
+|                          | tables TAB1,TAB2,TAB3                           |
+|                          | tables TAB1 <br/> tables TAB2,TAB3              |
+
+!!!note
+   The duplication of options is only supported for the options file. <br/>
+   When using `-tables` in the command line more than one time, only the last parameter is executed. For example, `./runMTK.sh -tables TAB2 -tables TAB3` only includes the _TAB3_ table in the migration.
+!!!
+
+### Use a space or equal sign to provide values
+
+For options that require values (key-value pairs), separate the option from the value with a space or an equal sign (=). Both options are supported.
+
+| Option in CLI    | Supported syntax for the option in file |
+|------------------|-----------------------------------------|
+| `-views all_emp` | views=all_emp                           |
+|                  | views all_emp                           |
+
+### Don't specify an option in the file and command line
+
+Don't include an option in the options file if you're specifying it as a flag in the command line.
+
+Specifying an option in both the command line and in the text file will cause the migration to fail.
+
+## Order of processing
+
+Migration Toolkit reads command line options and option files in the order you provide them when running the command.
+
+For example, if you run the following command, MTK first recognizes the `-sourcedbtype oracle` option, then reads the contents of `example.options` in order from top to bottom. Last parameter is always `schema_name`.
+
+```shell
+runMTK.sh -sourcedbtype oracle -optionsFile example.options schema_name
+```
+
+If you want an option to be executed last, you can either put it at the end of the `example.options` file and add no further options after `-optionsFile` in the command line.
+
+Schema definition (`schema_name` or `-allSchemas`) must always be the last parameter on the command line.
```

Comment:
When you specify a schema name or a comma separated list of schemas, those have to be the last argument of the command.  However, I don't think it matters where -allSchemas appears in the list of command options.
@zahid75 can you confirm?

Reply:
It does:

```
runMtk.sh -allSchemas -allTables
 
MTK-10026:-allSchemas option should be specified as the last option in the command-line options list
```

Reply:
For now, can we change the following:

> Schema definition (`schema_name` or `-allSchemas`) must always be the last parameter on the command line.

to:

> If migrating a subset of schemas, the comma separated list of schema names must be the last parameter on the command line.

I'd rather be silent on `-allSchemas` option if we are not certain that it needs to be specified last, which I don't think is true.

Reply:
I rescind my suggested change given @jk-edb confirmation that the option needs to be specified last.